Codeinwp/themeisle-sdk
# Optimole promotions

Adds Optimole promotions to all products using SDK.

The following places should show a notice:

- The main **Media** dashboard page;
- The **Attachment** modal in the **Media** dashboard page (for images that are larger than 200kb);
- After publishing a new post in the **Post Publish Panel** (for posts with two or more image blocks);
- Inside **Elementor** in the **Image Widget** content panel;

You can see some screenshots in the initial issue: https://github.com/Codeinwp/optimole-service/issues/608


---

The notices are randomly picked from a pool together with the ones for Otter blocks. They are then filtered based on generic conditions (if the plugin is installed or was installed via the notice and removed later, they won't show), then filtered based on the current page we're showing. From those remaining, we randomly pick a single one.

Once one of the notices is dismissed, another one should not show up again for 7 days.

To see which notice should show on the page you are, you can run the following in the browser console:

```js
themeisleSDKPromotions.showPromotion
```
![image](https://user-images.githubusercontent.com/15010186/201128958-6ec46a05-4eb0-4ab7-bf68-c14d561e242d.png)

---

There is a filter that can be added to enable **all the Optimole** notices (this won't also enable the Otter ones) and override the 7 days timeframe:

```php
add_filter('themeisle_sdk_promo_debug', '__return_true');
```

Here are some more WP-CLI commands that you might find useful during testing:

```sh
# Removes the option that holds all dismiss times for notices:
wp option delete themeisle_sdk_promotions    

# Removes the option that flagged Optimole as installed through any of the notices:
wp option delete themeisle_sdk_promotions_optimole_installed

# Removes the option that flagged Otter as installed through any of the notices:
wp option delete themeisle_sdk_promotions_otter 
```
 
### Testing this:

- You can use the dummy plugin in the comment below: https://github.com/Codeinwp/themeisle-sdk/pull/143#issuecomment-1311578974, which uses this version of the SDK;
- This plugin already has the filter mentioned above enabled, so if you want to disable it, you can delete it from the main plugin file;
- It will also load Otter promos, but not all of them simultaneously;
- Check that the notices appear in the correct places and that once they are dismissed, they don't appear again;
- Check that all Otter notices can be dismissed and they work fine;
- Check that the notices install and activate Optimole;
- If the account creation doesn't work fine (in case the email address is already registered with Optimole it will fail), the notice won't show an error, it just acts as if everything worked and shows a link to the Optimole plugin dashboard;

_Note: A known issue when using the filter to display all promos, is that when you insert an image in Elementor, the image widget promo disappears and it renders inside the attachment window. This happens because the attachment modal promo also gets loaded. In a production environment, the attachment promo will never load in the Elementor context._

I hope this makes sense. If there's any more need for clarification, please let me know! 👍🏻 

Closes https://github.com/Codeinwp/optimole-service/issues/608
